### PR TITLE
feat(engine)!: meter WASM execution and lift fee divisors into FeeTable

### PR DESCRIPTION
**Pull request by sdbondi**
## Description

Two related changes shipped together because the second depends on the first:

1. **WASM compute metering** — Wasmer's metering middleware was already wired into the runtime but the points were never converted to fees. This PR closes that gap: `WasmProcess::invoke` snapshots `get_remaining_points` before and after `func.call`, computes the delta (or the full pre-call budget when Wasmer traps on `Exhausted`), and charges the difference as a new `FeeSource::WasmExecution` line on the receipt. Failure paths bill the same as success paths.

2. **Divisor refactor** — Previously, three cost-scaling divisors were hard-coded magic numbers scattered through `FeeModule`. They are now first-class fields on `FeeTable`, making them network-configurable with a single table edit per deployment.

## Motivation

The fee model had no compute meter: a transaction could spin in a tight WASM loop for the full gas budget at zero marginal cost. The storage divisor (`4`) and template-load divisor (`3000`) were acknowledged in TODO comments as arbitrary; the only way to change them was a code deploy. Moving them into `FeeTable` makes calibration a one-liner — and now the hardware benchmark that is still owed turns into a one-line table edit rather than a code change.

Storage is charged at the full per-byte rate (`storage_cost_divisor = 1`) rather than a haircut: storage is the only indefinite-liability category in a receipt, so we treat it conservatively until a proper rent/endowment model lands.

## Changes

- `crates/engine_types/src/fees.rs` — new `FeeSource::WasmExecution` variant
- `crates/engine/src/runtime/module.rs` — new `RuntimeModule::on_wasm_execution(track, points)` hook
- `crates/engine/src/runtime/mod.rs`, `runtime/impl.rs` — new `RuntimeInterface::charge_wasm_execution` bridge
- `crates/engine/src/wasm/process.rs` — point snapshot + delta computation + charge in `WasmProcess::invoke`
- `crates/engine/src/fees/fee_module.rs` — `FeeModule::on_wasm_execution` implementation; references to divisor constants replaced with table lookups
- `crates/engine/src/fees/fee_table.rs` — four new fields (`per_wasm_point_cost`, `wasm_points_cost_divisor`, `template_load_bytes_cost_divisor`, `storage_cost_divisor`) plus `non_zero()` guard
- `applications/tari_ootle_app_utilities/src/fee_tables.rs` — network defaults for testnet and mainnet
- `crates/template_test_tooling/src/template_test.rs` — test fee table updated
- `crates/engine/tests/fees.rs` — fee-edge-case thresholds bumped (30/80/20 → 100/150/100) to account for WASM execution charge on built-in account invocations

## Calibration / follow-ups

The starting numbers (`per_wasm_point_cost = 1` µT per 1000 Wasmer points; `storage_cost_divisor = 1`) are educated-guess defaults. A hardware benchmark on the target validator spec is still owed to validate them. With this PR, that benchmark result is a one-line `FeeTable` edit per network — no code change required.

## Breaking changes

`FeeTable` gains four new required fields and `FeeSource` gains a new variant. Any code outside the workspace that constructs a `FeeTable` literal or matches `FeeSource` exhaustively will need to be updated.

## Test plan

- [x] Full workspace test suite passes: `cargo nextest r -E "not package(integration_tests)"` — 885/885
- [x] `cargo +nightly-2025-06-25 fmt --all` clean
- [x] Manual: submit a transaction and inspect the receipt — confirm a `WasmExecution` fee line appears with a sensible magnitude relative to `TemplateLoad` and `Storage` lines

### CODE REVIEW

File: crates/engine/src/fees/fee_module.rs [hunk 1]
```diff
@@ -134,4 +127,17 @@ impl<TStore: StateReader> RuntimeModule<TStore> for FeeModule {
 
         Ok(())
     }
+
+    fn on_wasm_execution(
+        &self,
+        track: &mut StateTracker<TStore>,
+        points_consumed: u64,
+    ) -> Result<(), RuntimeModuleError> {
+        let units = points_consumed / self.fee_table.wasm_points_cost_divisor();
```
[the remark this answers is not in the document]
Comment (sdbondi) [reply crates/engine/src/fees/fee_module.rs]: Good catch — fixed in e5963fad6.

Raw points are now accumulated on `FeeState` and the divisor is applied once in `on_before_finalize`, so 10 × 900-point calls correctly round to `9000 / 1000 = 9` units instead of `10 × (900/1000) = 0`. Matches how `Storage` already works (sum all bytes, divide once at finalize).
